hwilib/devices/coldcard.py

```diff
@@ -147,18 +149,43 @@ def sign_message(self, message, keypath):
 
     # Display address of specified type on the device. Only supports single-key based addresses.
     @coldcard_exception
-    def display_address(self, keypath, p2sh_p2wpkh, bech32):
-        self.device.check_mitm()
-        keypath = keypath.replace('h', '\'')
-        keypath = keypath.replace('H', '\'')
-
-        if p2sh_p2wpkh:
-            format = AF_P2WPKH_P2SH
-        elif bech32:
-            format = AF_P2WPKH
+    def display_address(self, keypath, p2sh_p2wpkh, bech32, redeem_script=''):
+        # redeem_script means p2sh/multisig
+        if redeem_script:
```

Comment:
Big if-statement based on `redeem_script` is ugly ...

New code below adapted from [here](https://github.com/Coldcard/ckcc-protocol/blob/master/ckcc/cli.py#L610)